File: app/schemas/retrieval.py
import uuid
from pydantic import BaseModel
from app.schemas.document import DocumentType


class RetrievalQuery(BaseModel):
    query: str
    document_type: DocumentType | None = None
    document_id: uuid.UUID | None = None
    top_k: int = 10


class RetrievedChunk(BaseModel):
    chunk_id: uuid.UUID
    document_id: uuid.UUID
    filename: str
    content: str
    chunk_type: str
    start_page: int
    end_page: int
    data_start_page: int | None = None
    data_end_page: int | None = None
    score: float

File: frontend/types/retrieval.ts
import type { DocumentType } from "@/types/chat";


export interface RetrievalQuery {
    query: string;
    document_type?: DocumentType;
    document_id?: string;
    top_k: number;
}


export interface RetrievedChunk {
    chunk_id: string;
    document_id: string;
    filename: string;
    content: string;
    chunk_type: string;
    start_page: number;
    end_page: number;
    data_start_page: number | null;
    data_end_page: number | null;
    score: number;
}